extensions/amp-auto-ads/0.1/ad-network-config.js

```diff
@@ -65,16 +67,16 @@ class AdSenseNetworkConfig {
 
   /** @override */
   getConfigUrl() {
+    const docInfo = documentInfoForDoc(this.autoAmpAdsElement_);
+    const canonicalHostname = parseUrl(docInfo.canonicalUrl).hostname;
```

Comment:
In terms of our backend systems, it's important that the hostname that we key the config by is the same as the hostname in the page URL that gets sent in the ad requests. This seems to be canonical URL from the code link I sent above.

A publisher actively providing a canonical URL with an incorrect hostname is something I guess we can't support.